beater/beater.go

```diff
@@ -30,21 +29,15 @@ func New(_ *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 }
 
 func (bt *beater) Run(b *beat.Beat) error {
-
 	var err error
 
-	bt.client, err = b.Publisher.Connect()
+	pub, err := newPublisher(b.Publisher)
 	if err != nil {
 		return err
 	}
-	defer bt.client.Close()
-
-	callback := func(events []beat.Event) {
-		// Publishing does not wait for publishing to be acked
-		go bt.client.PublishAll(events)
-	}
+	defer pub.Stop()
```

Comment:
- pub.Stop() blocks until the internal queue is drained. 
- the HTTP server must be shutdown before calling stop. Sending to `pub` after `Stop` generates a panic.

Reply:
would be useful to have these as comments

Reply:
@urso can you add a doc comment here?